Shared/Action/DataStoreAction.swift

```diff
@@ -16,6 +16,59 @@ enum DataStoreAction: Action {
     case sync
     case touch(id: String)
     case delete(id: String)
+    case syncEnd
+    case syncTimeout(error: String)
+    case syncError(error: String)
+}
+
+extension DataStoreAction: TelemetryAction {
+    var eventMethod: TelemetryEventMethod {
+        switch(self) {
+        case .updateCredentials:
+            return .update_credentials
+        case .lock:
+            return .lock
+        case .unlock:
+            return .unlock
+        case .reset:
+            return .reset
+        case .sync:
+            return .sync
+        case .touch:
+            return .touch
+        case .delete:
+            return .delete
+        case .syncEnd:
+            return .sync_end
+        case .syncTimeout:
+            return .sync_timeout
+        case .syncError:
+            return .sync_error
+        }
+    }
+
+    var eventObject: TelemetryEventObject {
+        return .datastore
+    }
+
+    var value: String? {
+        switch(self) {
+        case .delete(let id):
+            return id
+        case .touch(let id):
+            return "ID: \(id)"
+        case .syncError(let error):
```

Comment:
according the telemetry library there is a max string length of 50 https://github.com/mozilla-mobile/telemetry-ios/blob/aac1e1873134ea69f992e3bbc45b431d6564d64c/Telemetry/TelemetryEvent.swift#L8 its unclear to me what would happen if we pass it something longer,e.g.  if an error message is really long